.github/workflows/cargo-lint.yml

```diff
@@ -0,0 +1,49 @@
+name: Cargo lint and test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  # Not needed in CI, should make things a bit faster
+  CARGO_INCREMENTAL: 0
+  CARGO_TERM_COLOR: always
+  # Remove unnecessary WASM build artifacts
+  WASM_BUILD_CLEAN_TARGET: 1
+  # Stripping symbols and optimizing for binary size
+  RUSTFLAGS: -C strip=symbols -C opt-level=s
+  RUSTC_WRAPPER: sccache
+  SCCACHE_GHA_ENABLED: true
+
+jobs:
+  lint-and-test:
```

Comment:
(non-blocking)
I'm not entirely sure (it might be run as part of some of the existing commands), but it seems to me that this only runs tests and not lint.